jerry-core/lit/lit-magic-strings.c

```diff
@@ -183,7 +183,7 @@ lit_is_utf8_string_magic (const lit_utf8_byte_t *string_p, /**< utf-8 string */
   while (first <= last)
   {
     int middle = ((first + last) / 2); /**< mid point of search */
-    int compare = memcmp (lit_get_magic_string_utf8 (middle), string_p, string_size);
+    int compare = memcmp (lit_get_magic_string_utf8 ((lit_magic_string_id_t) middle), string_p, string_size);
 
```

Comment:
Sure that a cast is the right fix here? Casting should only be a last resort if there are no other options :)